pkg/plan/resource/kubeadm_init.go

```diff
@@ -251,13 +251,29 @@ func buildKubeadmInitPlan(path string, ignorePreflightErrors string, useIPTables
 		uploadCertsFlag = "--experimental-upload-certs"
 	}
 
+	// If we're at 1.17.0 or greater, we need to upgrade the kubeadm config before running "kubeadm init"
+	upgradeKubeadmConfig := false
+	if lt, err := version.LessThan(k8sVersion, "1.17.0"); err == nil && lt == false {
+		upgradeKubeadmConfig = true
+	}
+
 	b := plan.NewBuilder()
 	if useIPTables {
 		b.AddResource(
 			"configure:iptables",
 			&Run{Script: object.String("sysctl net.bridge.bridge-nf-call-iptables=1")}) // TODO: undo?
 	}
 
+	if upgradeKubeadmConfig {
+		b.AddResource(
```

Comment:
Does the `kubeadm:run-init` resource not depend on this one? Currently, we don't run these in parallel but we might go back to it in the future and the dependency graph might start to matter again.